test/test_transforms_tensor.py

```diff
@@ -497,7 +524,13 @@ def test_random_affine_fill(device, interpolation, fill):
 
 @pytest.mark.parametrize("device", cpu_and_gpu())
 @pytest.mark.parametrize("center", [(0, 0), [10, 10], None, (56, 44)])
-@pytest.mark.parametrize("expand", [True, False])
+# Most cases fail with expand=True, either a because of different values or even
+# because of a shape mismatch.
+# Needs to be verified but I think the shape mismatch is expected if the
+# rotation isn't exactly the same.
+# I assume it's because the random parameters are sampled differently between V1 and V2 (and so between V2 and jitted-V2)
+# Do we care?
```

Comment:
My original investigation about this in https://github.com/pytorch/vision/pull/7159#issuecomment-1423212817 is wrong. What we are actually hitting is

- v1:
  https://github.com/pytorch/vision/blob/3991ab994630ede87706dc9b46d005b8834cc5bc/torchvision/transforms/functional.py#L1131-L1135
- v2:
  https://github.com/pytorch/vision/blob/3991ab994630ede87706dc9b46d005b8834cc5bc/torchvision/transforms/v2/functional/_geometry.py#L832-L838

The warning in v2 is on point: with `expand=True` the result will always be the same for an arbitrary `center`. However, this is only true in theory. In practice, keeping the center for the matrix computation leads to slight numerical errors that ultimately manifest as +-1 in the computed shape.

The check was added in #5638. I'm ok with keeping this in, but no strong opinion. @vfdev-5?

Reply:
Thansk for digging Philip. My brain is close to be fried already but why do we have an `if / else` in v2?

I undertand and agree with the warning, but shouldn't we **still** update `center_f` regardless of whether `expand` was passed? i.e.

```py
 if center is not None: 
     if expand: 
         warnings.warn("The provided center argument has no effect on the result if expand is True") 
    # NO ELSE !!!
     # Center values should be in pixel coordinates but translated such that (0, 0) corresponds to image center. 
     center_f = [(c - s * 0.5) for c, s in zip(center, [width, height])] 

```

Reply:
ah ok - brain is positively fried but it looks like both snippets are correct. Maybe the one without the `else` would lead to the same results as in V1?

Reply:
I also wonder whether we should actually have that warning or just make it part of the docstring. It can be annoying when swiping through different configurations?

Reply:
Fine for me. Previously it was a bug fix and now it will be a full BC behaviour. I'm ok with that

Reply:
Done in https://github.com/pytorch/vision/pull/7271 - tests are OK now. We still have some sporadic random failures on some tests when expand=True but it's only ~1 pixel off, not different shapes. Thanks al lot both for your help